robot/impl/robot_reconfigure_test.go

```diff
@@ -320,11 +326,10 @@ func TestRobotReconfigure(t *testing.T) {
 
 		rr.triggerConfig <- struct{}{}
 
-		utils.SelectContextOrWait(context.Background(), 200*time.Millisecond)
-
-		_, err = robot.ResourceByName(mockNamed("mock2"))
-		test.That(t, err, test.ShouldBeNil)
-
+		testutils.WaitForAssertionWithSleep(t, time.Millisecond*100, 20, func(tb testing.TB) {
```

Comment:
Because this is not the helper function, but rather a function I'm passing TO the helper function, which is testutils.WaitForAssertionWithSleep itself (as defined in goutils) which sets t.Helper() itself.

Marking something a helper is when you have, say, a library function that does some complicated thing, but you want it to appear as only one test (and give the line numbers of only the entrance to the helper) when it fails. E.g. you don't want to see code and lines from the utility library.

I definitely used to be confused by this myself, and there's probably some cases where I added t.Helper() wrongly elsewhere in RDK still. :-)